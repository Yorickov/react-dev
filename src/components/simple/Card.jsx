import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ title, text }) => (
  <div className="card">
    <div className="card-body">
      <h4 className="card-title">{title}</h4>
      <p className="card-text">{text}</p>
      <button type="button" className="btn btn-primary">Press me</button>
    </div>
  </div>
);

Card.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
};

export default Card;
