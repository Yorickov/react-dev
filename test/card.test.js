import React from 'react';
import Card from '../src/components/simple/Card.jsx';

it('Card', () => {
  const params = {
    title: 'title',
    text: 'text',
  };
  const wrapper = shallow(<Card { ...params } />);
  expect(wrapper.render()).toMatchSnapshot();
});
