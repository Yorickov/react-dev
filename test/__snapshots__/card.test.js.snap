// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 1`] = `
<div
  class="card"
>
  <div
    class="card-body"
  >
    <h4
      class="card-title"
    >
      title
    </h4>
    <p
      class="card-text"
    >
      text
    </p>
    <button
      class="btn btn-primary"
      type="button"
    >
      Press me
    </button>
  </div>
</div>
`;
